include/openthread/border_agent.h

```diff
@@ -299,6 +299,23 @@ void otBorderAgentSetEphemeralKeyCallback(otInstance                       *aIns
                                           otBorderAgentEphemeralKeyCallback aCallback,
                                           void                             *aContext);
 
+/**
+ * Starts the Border Agent service.
+ *
+ * @param[in] aInstance    The OpenThread instance.
+ *
+ */
+void otBorderAgentStart(otInstance *aInstance);
```

Comment:
I think we need a `otBorderAgentStEnabled(bool)` API here to allow enabling/disabling the Border Agent feature. `otBorderAgentStop()` doesn't always stops the border agent because it will be automatically started when the device state becomes attached. `otBorderAgentStEnabled(bool)` is also useful for Thread in mobile use case where the device is not a border router but we need to use the same binary for both border router and non-border-router.

Reply:
+1 on adding `SetEnabled()` as an overall mechanism to enable/disable the BA function.

The BA is indeed auto-started/stopped from `BorderAgent::HandleNotifierEvents()` on role and commissioner  change events.

----

Regarding the main use-case here, how about introducing a new `otBorderAgentDisconnect()` API to disconnect any active secure session.

Thoughts?

Reply:
`otBorderAgentDisconnect ()` sounds good, created a PR for it: https://github.com/openthread/openthread/pull/10754